packages/aws-cdk/lib/cli.ts

```diff
@@ -262,7 +262,8 @@ async function parseCommandLineArguments(args: string[]) {
       .option('security-only', { type: 'boolean', desc: 'Only diff for broadened security changes', default: false })
       .option('fail', { type: 'boolean', desc: 'Fail with exit code 1 in case of diff' })
       .option('processed', { type: 'boolean', desc: 'Whether to compare against the template with Transforms already processed', default: false })
-      .option('quiet', { type: 'boolean', alias: 'q', desc: 'Do not print stack name and default message when there is no diff to stdout', default: false }))
+      .option('quiet', { type: 'boolean', alias: 'q', desc: 'Do not print stack name and default message when there is no diff to stdout', default: false })
+      .option('change-set', { type: 'boolean', desc: 'Whether to create a changeset to analyze resource replacements', default: true }))
```

Comment:
This uses the deploy role to do it -- did the previous `GetTemplate` also use the deploy role? (If yes, nothing to do, but otherwise you probably need to add that in as a warning)

Reply:
Also add in that this is slower, and make sure we print a message while we're waiting for `createchangeset` to happen. It can easily take 10-30s, and previously `cdk diff` was nearly instantaneous.